Assets/Scripts/Framework/PointsSystem/PointsTextUpdater.cs

```diff
@@ -0,0 +1,29 @@
+using TMPro;
+using UnityEngine;
+
+namespace ProefExamen.Framework.PointsSystem
+{
+    /// <summary>
+    /// A class responsible for updating the points text.
+    /// </summary>
+    [RequireComponent(typeof(TextMeshProUGUI))]
+    public class PointsTextUpdater : MonoBehaviour
+    {
+        private TextMeshProUGUI _pointsText;
+        private string _defaultPointsText;
+
+        private void Awake()
+        {
+            _pointsText = GetComponent<TextMeshProUGUI>();
+            _defaultPointsText = _pointsText.text;
+        }
+
+        private void Start() => UpdatePointsText(PointsSystem.Points);
+
+        private void OnEnable() => PointsSystem.OnPointsChanged += UpdatePointsText;
+
+        private void OnDisable() => PointsSystem.OnPointsChanged -= UpdatePointsText;
+
+        private void UpdatePointsText(int points) => _pointsText.text = _defaultPointsText.Replace("[points]", points.ToString());
```

Comment:
This lamda one liner also looks a bit long, if it is less than 120 characters, I would maybe suggest we look at the convention about the one-liners. 

From a glance it's a bit hard to read what it does. Maybe it's just me but right here, I would prefer this 👇
(also since this would be easier to debug if needed with break points)

```
private void UpdatePointsText(int points) 
    => _pointsText.text = _defaultPointsText.Replace("[points]", points.ToString());
```

Reply:
This one is indeed over the 120 characters limit. I will change it. Maybe we can discuss a change in the code convention after the break.